src/rest/login.ts

```diff
@@ -0,0 +1,31 @@
+import { getUserByUsernameAndHash } from "../db/User/user";
+import { Request, Response } from "express";
+import { AuthenticationError } from "../types";
+import { createHash } from "crypto";
+import StatusCode from "./status";
+
+export const login = async (
+	req: Request<unknown, unknown, { username: string; hash: string }>,
+	res: Response
+): Promise<void> => {
+	const { username, hash } = req.body;
+
+	try {
+		await getUserByUsernameAndHash(username, hash);
+	} catch (e) {
+		if (e instanceof AuthenticationError) {
+			res.status(StatusCode.NOT_FOUND).send("Account not found");
+			return;
+		} else {
+			throw e;
+		}
+	}
+
+	const token = createHash("sha256")
+		.update(`${new Date().toISOString()}${process.env.AUTH_SEED}`)
+		.digest("hex");
+	const tokenStore: Record<string, string> = req.app.get("tokenStore") as Record<string, string>;
+	tokenStore[username] = token;
+	req.app.set("tokenStore", tokenStore);
+	res.status(StatusCode.OK).send(token);
```

Comment:
so the user gets back their token, and then in follow up requests like create-rating, they will send this token as a header for example, but then what? We look up the token in the tokenStore by value to find the username and get that user from the DB?

Reply:
yes, or we the header sends the token and the username, verifies that the token matches the username, and only creates the rating if there's a match